frank/radial_fitters.py

```diff
@@ -922,6 +923,10 @@ def MAP_spectrum(self):
     @property
     def MAP_spectrum_covariance(self):
         """Covariance matrix of the maximum a posteriori power spectrum"""
+        if self._ps_cov is None:
```

Comment:
Did you test this? Can you confirm it works for me.

Reply:
I confirm it returns `MAP_spectrum_covariance` as, e.g.,

```
Message: 'FF.MAP_spectrum_covariance'
Arguments: (array([[ 1.81711780e+00,  2.11017083e-03, -1.39366320e-03, ...,
         2.91032411e-07,  3.09998364e-07,  3.31108797e-07],
       [ 2.11017083e-03,  1.81184208e+00,  1.46379719e-02, ...,
        -7.32676623e-07, -7.79869609e-07, -8.32481953e-07],
       [-1.39366320e-03,  1.46379719e-02,  1.85701394e+00, ...,
         1.05319287e-06,  1.12157499e-06,  1.19773501e-06],
       ...,
       [ 2.91032411e-07, -7.32676623e-07,  1.05319287e-06, ...,
         7.97298342e-01,  8.14643340e-01,  8.28254145e-01],
       [ 3.09998364e-07, -7.79869609e-07,  1.12157499e-06, ...,
         8.14643340e-01,  8.94548276e-01,  9.69512558e-01],
       [ 3.31108797e-07, -8.32481953e-07,  1.19773501e-06, ...,
         8.28254145e-01,  9.69512558e-01,  1.11490208e+00]]),)
```

but I haven't checked beyond that; this seems to me sufficient though.